warehouse/templates/accounts/two-factor.html

```diff
@@ -104,12 +104,20 @@ <h2>{% trans %}Authenticate with an app{% endtrans %}</h2>
 
           {% trans trimmed href='/help/#totp' %}
           <p>Generate a code using the authentication application connected to your PyPI account. Enter this code in the form to verify your identity.</p>
-          <p>Lost your application? Not working? <a href="{{ href }}">Get help</a>.</p>
+          <p>Lost your application?  Not working? <a href="{{ href }}">Get help</a>.</p>
           {% endtrans %}
         </form>
       </div>
       {% endif %}
     </div>
+
+    <hr>
+    <div class="twofa-recovery">
+      <p>{% trans %}Lost your security key or application?{% endtrans %}</p>
+      <a href="{{ request.route_path('accounts.recovery-code') }}?{{request.query_string}}" class="button button--secondary">
+          {% trans %}Login using a Recovery Code{% endtrans %}
+      </a>
+    </div>
```

Comment:
This is displayed whether or not the user has recovery codes. Perhaps intentionally? But this might be a good opportunity to direct a user that does have 2FA but hasn't generated recovery codes that they should generate codes once logging in.